lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -0,0 +1,635 @@
+            :param hostnames: a list of hostname destination variables in order of preference
+        '''
+        for host in hosts:
+            hostname = self._get_hostname(host, hostnames)
+            self.inventory.add_host(hostname, group=group)
+            for hostvar in host.instance_data.keys():
+                self.inventory.set_variable(hostname, hostvar, host.instance_data[hostvar])
+
+    def _find_cred(self, possible_env_vars):
+        '''
+            :param possible_env_vars: the possible environment variables to check for a credential
+        '''
+        for env_var in possible_env_vars:
+            if os.environ.get(env_var):
+                return os.environ.get(env_var)
+
+    def _set_credentials(self, config_data):
+        '''
+            :param config_data: contents of the inventory config file
+        '''
+
+        self.boto_profile = config_data.get('boto_profile')
+        self.aws_access_key_id = config_data.get('aws_access_key_id')
+        self.aws_secret_access_key = config_data.get('aws_secret_access_key')
+        self.aws_security_token = config_data.get('aws_security_token')
+
+        if not self.boto_profile:
+            self.boto_profile = self._find_cred(possible_env_vars=('AWS_PROFILE',
+                                                                   'AWS_DEFAULT_PROFILE'))
+
+        if not self.aws_access_key_id:
+            self.aws_access_key_id = self._find_cred(possible_env_vars=('AWS_ACCESS_KEY_ID',
+                                                                        'AWS_ACCESS_KEY',
+                                                                        'EC2_ACCESS_KEY'))
+
+        if not self.aws_secret_access_key:
+            self.aws_secret_access_key = self._find_cred(possible_env_vars=('AWS_SECRET_ACCESS_KEY',
+                                                                            'AWS_SECRET_KEY',
+                                                                            'EC2_SECRET_KEY'))
+
+        if not self.aws_security_token:
+            self.aws_security_token = self._find_cred(possible_env_vars=('AWS_SECURITY_TOKEN',
+                                                                         'AWS_SESSION_TOKEN',
+                                                                         'EC2_SECURITY_TOKEN'))
+        if not self.boto_profile and not (self.aws_access_key_id and self.aws_secret_access_key):
+            raise AnsibleError("Insufficient boto credentials found. Please provide them in your "
+                               "inventory configuration file or set them as environment variables.")
+
+    def _validate_config(self, loader, path):
+        '''
+            :param loader: an ansible.parsing.dataloader.DataLoader object
+            :param path: the path to the inventory config file
+            :return the contents of the config file
+        '''
+        self.verify_file(path)
```

Comment:
Is this call redundant? The manager already calls this earlier, so seems like it shouldn't need to happen again (plus we're ignoring its result).